src/state/secret.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""gateway-api-integrator secret resource definition."""
+
+import dataclasses
+
+import ops
+
+
+@dataclasses.dataclass(frozen=True)
+class SecretResourceDefinition:
+    """A component of charm state that contains secret resource definition.
+
+    Attrs:
+        secret_resource_name_prefix (str):
```

Comment:
Perhaps the `(str)` is not required here :D

Reply:
I updated the docstring here, thanks!